packages/core/src/configuration/Configuration.ts

```diff
@@ -0,0 +1,76 @@
+import { CredentialProvider, Credentials } from '../credentials-provider';
+import { container, DependencyContainer, injectable } from 'tsyringe';
+
+export interface ConfigurationOptions {
+  credentials?: Credentials;
+  cluster: string;
+  credentialProviders?: CredentialProvider[];
+}
+
+@injectable()
+export class Configuration {
+  public readonly credentialProviders?: CredentialProvider[];
```

Comment:
You should hide everything that allows modifying array from outside

```suggestion
  private _credentialProviders?: CredentialProvider[];
  
  get credentialProviders(): readonly CredentialProvider[] {
    return this._credentialProviders;
  }
```